src/main/java/bookstore/mapper/BookMapper.java

```diff
@@ -24,4 +25,6 @@ public interface BookMapper {
 
     @BeanMapping(nullValuePropertyMappingStrategy = NullValuePropertyMappingStrategy.IGNORE)
     void updateBook(CreateBookRequestDto requestDto, @MappingTarget Book book);
+
+    BookDtoWithoutCategoryIds toDtoWithoutCategories(Book book);
```

Comment:
Component annotation here is redundant. Also, BookDto should return the response with categoryIds 
You also should provide possibility add  a book with categories